manifests/init.pp

```diff
@@ -284,84 +284,119 @@
       $package_name_default      = 'nscd'
       $package_source_default    = undef
       $service_name_default      = 'nscd'
-      case $::operatingsystemmajrelease {
-        '5': {
```

Comment:
I'm not sure that we will be able to get known which Amazon version corresponds to EL version. Amazon doesn't provide such information. 
It used to be something between EL5 and 6:
[Amazon forum year: 2010](https://forums.aws.amazon.com/message.jspa?messageID=194752) 
And now according to version of some packages it's closer to EL 7.

Using Amazon linux  major version ("2016", "2015") in RedHat section isn't obvious imho.